### PR TITLE
[Bug] Hotfix

### PR DESCRIPTION
**Pull request by wadhawh**
This PR includes the following;

- Removed dual reload for a new user
- Fixes error page redirection for new user
- Checks App version before rendering App
- Checks identityPoolId only when demo app is accessed
- Removed unnecessary routes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/core/constants/appConfig.ts [hunk 1]
```diff
@@ -83,19 +83,15 @@ const appConfig = {
 	ROUTES: {
 		DEFAULT: "/",
 		DEMO: "/demo",
+		NL_DEMO: "/demo?dp=Esri&nl=true",
```
Comment (imraymondlee) [inline src/core/constants/appConfig.ts:86]: What is this used for and have we implemented this yet?

File: src/setupTests.ts [hunk 2]
```diff
@@ -40,35 +39,36 @@ jest.mock("@demo/core/constants/appConfig", () => ({
...
 		HELP: "/demo/help",
 		TERMS: "/demo/terms",
-		SOFTWARE_ATTRIBUTIONS: "/showcase/software-attributions",
-		BLOG: "/blog",
-		NEW: "/new",
-		UNAUTHORIZED: "/401",
-		FORBIDDEN: "/403",
-		NOT_FOUND: "/404",
-		ERROR_BOUNDARY: "/error"
+		SOFTWARE_ATTRIBUTIONS: "/showcase/software-attributions"
```
Comment (imraymondlee) [inline src/setupTests.ts:71]: Nit: the value in appConfig.ts is "/demo/software-attributions"

**Review by imraymondlee — APPROVED** [review]

File: src/setupTests.ts (hunk 2 above)
Comment (wadhawh) [reply src/setupTests.ts:71]: thanks @imraymondlee , fixing this right away

File: src/core/constants/appConfig.ts (hunk 1 above)
Comment (wadhawh) [reply src/core/constants/appConfig.ts:86]: This is used to redirect user to the NL search, this is implemented but controlled via an env flag

**Pull request merged**